internal/cli/auth/login_test.go

```diff
@@ -69,6 +69,43 @@ func TestLoginBuilder(t *testing.T) {
 	)
 }
 
+func Test_loginOpts_SyncWithOAuthAccessProfile_DefaultService(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockConfig := mocks.NewMockLoginConfig(ctrl)
+
+	opts := &LoginOpts{
+		NoBrowser:    true,
+		AccessToken:  "at",
+		RefreshToken: "rt",
+	}
+	opts.OutWriter = new(bytes.Buffer)
+
+	mockConfig.EXPECT().Set("service", "cloud").Times(1)
+	mockConfig.EXPECT().Set("access_token", opts.AccessToken).Times(1)
+	mockConfig.EXPECT().Set("refresh_token", opts.RefreshToken).Times(1)
+	mockConfig.EXPECT().Set("ops_manager_url", gomock.Any()).Times(0)
+	require.NoError(t, opts.SyncWithOAuthAccessProfile(mockConfig)())
+}
+
+func Test_loginOpts_SyncWithOAuthAccessProfile_GovService(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockConfig := mocks.NewMockLoginConfig(ctrl)
+
+	opts := &LoginOpts{
+		NoBrowser:    true,
+		AccessToken:  "at",
+		RefreshToken: "rt",
+		IsGov:        true,
+	}
+	opts.OutWriter = new(bytes.Buffer)
+
+	mockConfig.EXPECT().Set("service", "cloudgov").Times(1)
+	mockConfig.EXPECT().Set("access_token", opts.AccessToken).Times(1)
+	mockConfig.EXPECT().Set("refresh_token", opts.RefreshToken).Times(1)
+	mockConfig.EXPECT().Set("ops_manager_url", gomock.Any()).Times(0)
+	require.NoError(t, opts.SyncWithOAuthAccessProfile(mockConfig)())
+}
```

Comment:
[nit] in go you usually nest scenarios via `t.Run` instead of multple test definitions, check https://github.com/mongodb/mongodb-atlas-cli/blob/baec0342e394d1d57db60bafb14dd3e477487c4e/internal/cli/atlas/clusters/create_test.go#L30-L107 for one approach or https://github.com/mongodb/mongodb-atlas-cli/blob/baec0342e394d1d57db60bafb14dd3e477487c4e/internal/cli/global_opts_test.go#L26-L60 for a more advanced approach using a test matrix

Reply:
thank you!
refactored tests with second option.